Platforms/QemuQ35Pkg/QemuQ35Pkg.dsc

```diff
@@ -463,7 +463,7 @@
   PcdLib                        |MdePkg/Library/DxePcdLib/DxePcdLib.inf
   HobLib                        |MdePkg/Library/DxeHobLib/DxeHobLib.inf
   ResetSystemLib                |MdeModulePkg/Library/DxeResetSystemLib/DxeResetSystemLib.inf
-  HwResetSystemLib              |MdeModulePkg/Library/DxeResetSystemLib/DxeResetSystemLib.inf
+  HwResetSystemLib              |QemuQ35Pkg/Library/ResetSystemLib/DxeResetSystemLib.inf

```

Comment:
nit: I believe this is functionally fine. But not sure why we need this DXE specific instance. Seems to be redundant complication.

Reply:
Are you talking about why QemuQ35Pkg has a DxeResetSystemLib in addition to BaseResetSystemLib (instead of just BaseResetSystemLib)?

Reply:
Yes. The implementation only seems to differ by caching a global variable (for some cloud hv flavor).

Reply:
Yeah, I'm not sure of the history there. Based on what I see, I'm fine to update this to use all Base instances. Do you want to go with that?

Reply:
Yeah, let's do that. Thanks!

Reply:
I suspected it might have something to do with DXE runtime drivers due to the global variable caching. It does - https://github.com/tianocore/edk2/commit/93f6df5f3b2553b8f5188d2a6ba70f3f5cfab0bb.

Given that, I just updated the general DXE section to use the BASE instance. It's overridden for DXE Runtime later and I left the DXE instance for SMM and Standalone MM to keep the instance used across the runtime transition consistent.

Reply:
Makes sense. Thanks for looking into it!